test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -0,0 +1,436 @@
+ * DomainParticipant.
+ * 3. Use both narrow methods to obtain the pointer to the children class
+ * 4. Call both narrow methods with an invalid pointer and check that it returns nullptr
+ * 5. Delete DDS entities
+ */
+TEST(StatisticsDomainParticipantTests, NarrowDomainParticipantTest)
+{
+    eprosima::fastdds::dds::DomainParticipant * participant =
+            eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+            create_participant(0, eprosima::fastdds::dds::PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(participant, nullptr);
+
+#ifndef FASTDDS_STATISTICS
+    eprosima::fastdds::statistics::dds::DomainParticipant * statistics_participant =
+            eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant);
+    EXPECT_EQ(statistics_participant, nullptr);
+
+    ASSERT_TRUE(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+#else   
+    logError(STATISTICS_DOMAINPARTICIPANT_TEST, "This test is going to fail because API is not yet implemented.")
+
+    eprosima::fastdds::statistics::dds::DomainParticipant * statistics_participant =
+            eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant);
+    EXPECT_NE(statistics_participant, nullptr);
+
+    const eprosima::fastdds::dds::DomainParticipant * const_participant = participant;
+    const eprosima::fastdds::statistics::dds::DomainParticipant * const_statistics_participant =
+            eprosima::fastdds::statistics::dds::DomainParticipant::narrow(const_participant);
+    EXPECT_NE(const_statistics_participant, nullptr);
+
+    participant = nullptr;
+    statistics_participant = eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant);
+    EXPECT_EQ(statistics_participant, nullptr);
+
+    const_participant = nullptr;
+    const_statistics_participant = eprosima::fastdds::statistics::dds::DomainParticipant::narrow(const_participant);
+    EXPECT_EQ(const_statistics_participant, nullptr);
+
+    ASSERT_TRUE(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(statistics_participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_ERROR);
+#endif // FASTDDS_STATISTICS
+}
+
+/*
+ * This test checks both eprosima::fastdds::statistics::dds::DomainParticipant enable_statistics_datawriter() and
+ * disable_statistics_datawriter() methods.
+ * 1. If the FASTDDS_STATISTICS compilation flag is not set, check that they return RETCODE_UNSUPPORTED.
+ * 2. Create a eprosima::fastdds::dds::DomainParticipant and narrow to the children class.
+ * 3. Enable every single statistics datawriter successfully returning RETCODE_OK.
+ * 4. Check that the corresponding topic has been created and the type has been registered.
+ * 5. Enable and already enabled statistics DataWriter and check that it returns RETCODE_OK.
```

Comment:
```suggestion
 * 5. Enable an already enabled statistics DataWriter and check that it returns RETCODE_OK.
```

Reply:
Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3